docs/api/QUIC_STREAM_EVENT.md

```diff
@@ -0,0 +1,266 @@
+        struct {
+            QUIC_UINT62 ErrorCode;
+        } PEER_RECEIVE_ABORTED;
+        struct {
+            BOOLEAN Graceful;
+        } SEND_SHUTDOWN_COMPLETE;
+        struct {
+            BOOLEAN ConnectionShutdown;
+            BOOLEAN AppCloseInProgress       : 1;
+            BOOLEAN ConnectionShutdownByApp  : 1;
+            BOOLEAN ConnectionClosedRemotely : 1;
+            BOOLEAN RESERVED                 : 5;
+            QUIC_UINT62 ConnectionErrorCode;
+            QUIC_STATUS ConnectionCloseStatus;
+        } SHUTDOWN_COMPLETE;
+        struct {
+            uint64_t ByteCount;
+        } IDEAL_SEND_BUFFER_SIZE;
+        struct {
+            /* out */ QUIC_UINT62 ErrorCode;
+        } CANCEL_ON_LOSS;
+    };
+} QUIC_STREAM_EVENT;
+```
+
+# Parameters
+
+`Type`
+
+The `QUIC_STREAM_EVENT_TYPE` that indicates which type of event this is, and which payload to reference (if any) for additional information.
+
+# Remarks
+
+## QUIC_STREAM_EVENT_START_COMPLETE
+
+This event is delivered when the [StreamStart](streamStart.md) operation completes. The accompanying payload contains data to indicate whether the operation succeeded or failed.
+
+### START_COMPLETE
+
+Additional state from the `StreamStart` operation is included in this payload struct/union.
+
+`Status`
+
+QUIC_STATUS value to indicate the operation completion code. Check for success using the QUIC_SUCCEEDED macro.
+
+`ID`
+
+Stream ID if available.
+
+`PeerAccepted`
+
+If TRUE, the peer has accepted the stream.
+
+## QUIC_STREAM_EVENT_RECEIVE
+
+Data received on an open stream is primarily delivered to the application through this event.
+
+### RECEIVE
+
+Received data on the stream is passed in this struct/union.
+
+`AbsoluteOffset`
+
+Absolute offset of the current data payload from the start of the receive operation.
+
+`TotalBufferLength`
+
+MsQuic indicates the total buffer length of the data in this parameter.
+
+If the application can processes only part of the received buffer synchronously in this call, it can be indicated to MsQuic by setting this parameter to the byte count processed and returning `QUIC_STATUS_CONTINUE` from this call.
+If the application desires to process the received data asynchronously, it should return `QUIC_STATUS_PENDING` from this call.
+
+See [Receiving Data On Streams](../Streams.md#Receiving) for further details on receiving data on a stream.
+
+`Buffers`
+
+An array of `QUIC_BUFFER`s containing received data.
+
+`BufferCount`
+
+Count of `QUIC_BUFFER`s in this payload.
+
+`Flags`
+
+A set of flags indicating describing the received data:
+
+Value | Meaning
+--- | ---
+**QUIC_RECEIVE_FLAG_NONE**<br>0 | No special behavior.
+**QUIC_RECEIVE_FLAG_0_RTT**<br>1 | The data was received in 0-RTT.
+**QUIC_RECEIVE_FLAG_FIN**<br>2 | FIN was included with this data. Used only for streamed data.
+
+## QUIC_STREAM_EVENT_SEND_COMPLETE
+
+Indicates that MsQuic has completed a [StreamSend](StreamSend.md) operation initated by the application.
```

Comment:
Can we add here that it means the app regain ownership of the `QUIC_BUFFER` and of the referenced memory used for the matching "send" call, and can now free it / re-use it?

We recently got some questions about it.